c-api/compat-5.3.c

```diff
@@ -336,22 +392,155 @@ COMPAT53_API void luaL_traceback (lua_State *L, lua_State *L1,
 
 
 COMPAT53_API int luaL_fileresult (lua_State *L, int stat, const char *fname) {
+  const char* s = NULL;
   int en = errno;  /* calls to Lua API may change this value */
   if (stat) {
     lua_pushboolean(L, 1);
     return 1;
   }
   else {
     lua_pushnil(L);
+#if defined(__GLIBC__) || defined(_POSIX_VERSION) || defined(__APPLE__) || (defined(__GNUC__) && (__GNUC__ < 6) && defined (__MINGW32__))
+    // use strerror_r here, because it's available on these specific platforms
+    char buf[512];
+    strerror_r(en, buf, 1024);
+    s = buf;
+#elif defined(_MSC_VER)
+    // for MSVC and others, use strerror_s since it's provided by default by the libraries
+    char buf[512];
+    strerror_s(buf, 1024, en);
+    s = buf;
+#else
+    s = strerror(en);
+#endif
     if (fname)
-      lua_pushfstring(L, "%s: %s", fname, strerror(en));
+      lua_pushfstring(L, "%s: %s", fname, s);
     else
-      lua_pushstring(L, strerror(en));
+      lua_pushstring(L, s);
     lua_pushnumber(L, (lua_Number)en);
     return 3;
   }
 }
 
+static const char* compat53_f_parser_handler (lua_State *L, void *data, size_t *size) {
+  struct compat53_f_parser_buffer *p = (struct compat53_f_parser_buffer*)data;
+  size_t readcount = fread(p->buffer + p->start, sizeof(*p->buffer), sizeof(p->buffer), p->file);
+  if (ferror(p->file) != 0) {
+    *size = 0;
+    return NULL;
+  }
+  p->start = 0;
+  *size = readcount;
+  return p->buffer;
+}
+
+COMPAT53_API int luaL_loadfilex(lua_State *L, const char *filename, const char *mode) {
```

Comment:
I'd prefer that we copied lua 5.3's actual implementation. see http://www.lua.org/source/5.3/lauxlib.c.html#luaL_loadfilex

Reply:
That's fine, too: I can remove my implementation from the PR. I'm not exactly confident I can copy-paste that exactly as I'm not familiar with the Lua sources.